### PR TITLE
Adds CanvasGraph

### PR DESCRIPTION
**Pull request by jbrundage**
@GordonSmith please review

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "c3": "~0.4.11",
     "colorbrewer": "~1.0.0",
     "d3": "~3.5.17",
+    "d3v4": "https://d3js.org/d3.v4.min.js",
```
Comment (GordonSmith) [inline bower.json]: I keep forgetting that this graph is dependent on d3v4 - how hard would it be to use d3v3 for the 1.20.x branch and then upgrade it to d3v4 for the 2.0.x?

File: src/config.js [hunk 2]
```diff
@@ -41,100 +42,100 @@ require.config({
...
+                "simpleheat": "//" + window.location.hostname + "/mourner/simpleheat/v0.3.0/simpleheat",
+                "autoComplete": "//" + window.location.hostname + "/Pixabay/JavaScript-autoComplete/1.0.4/auto-complete",
 
-            "src": "../src"
-        }
-    });
+                "src": "../src"
+            }
+        });
 }
```
Comment (GordonSmith) [inline src/config.js]: Try to avoid gratutious white space changes (or make them on a separate commit with no "code" changes",

File: src/graph/CanvasGraphLayouts.js [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+        data.links.forEach(function (u) {
+            g.setEdge(u.source.index, u.target.index);
+        });
+        return g;
+    }
+
+    return {
+        layouts: layouts
+    };
+}));
```
Comment (GordonSmith) [inline src/graph/CanvasGraphLayouts.js]: Looks like duplicate code with the existing graph layouts?

File: src/graph/CanvasWidget.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    CanvasWidget.prototype.click = function(d,c) {
+        console.log(d);
+    };
+    
+    CanvasWidget.prototype.lazyRender = CanvasWidget.prototype.debounce(function () {
+        this.render();
+    }, 100);
+
+    return CanvasWidget;
+}));
```
Comment (GordonSmith) [inline src/graph/CanvasWidget.js]: Don't we already have a canvas widget in Common?

File: bower.json (hunk 1 above)
Comment (jbrundage) [reply bower.json]: I'm not sure how hard it would be without diving into it.

File: src/graph/CanvasGraphLayouts.js (hunk 3 above)
Comment (jbrundage) [reply src/graph/CanvasGraphLayouts.js]: Yes, some duplicate code. V2 should be able to share layout code with SVG graph.

File: src/graph/CanvasWidget.js (hunk 4 above)
Comment (jbrundage) [reply src/graph/CanvasWidget.js]: Yes, infact this file isn't even used... I'll remove it.

**Comment by jbrundage** [comment]
Closing in favor of #2413
